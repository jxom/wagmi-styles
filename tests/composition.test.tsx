import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import { themeA, themeB, style, ThemeProvider, useStyle } from './theme';

describe('composition', () => {
  const size = style((vars) => ({
    height: vars.sizes.small,
    width: vars.sizes.medium,
  }));
  const spacing = style({ padding: 16, margin: 8 });
  const composedStyleRef = style([
    size,
    spacing,
    {
      backgroundColor: 'red',
    },
  ]);

  function Composed() {
    const composedStyle = useStyle(composedStyleRef);
    return <View style={composedStyle} />;
  }

  [themeA, themeB].forEach((theme) => {
    test('composed style', () => {
      const { toJSON } = render(
        <ThemeProvider theme={theme}>
          <Composed />
        </ThemeProvider>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });

  ///////////////////////////////////////////////////////////////

  const composedStyleWithThemeVarsRef = style((vars) => [
    size,
    spacing,
    {
      backgroundColor: vars.colors.primary,
    },
  ]);

  function ComposedWithVars() {
    const composedWithVarsStyle = useStyle(composedStyleWithThemeVarsRef);
    return <View style={composedWithVarsStyle} />;
  }

  [themeA, themeB].forEach((theme) => {
    test('composed style with theme vars', () => {
      const { toJSON } = render(
        <ThemeProvider theme={theme}>
          <ComposedWithVars />
        </ThemeProvider>
      );
      expect(toJSON()).toMatchSnapshot();
    });
  });
});
