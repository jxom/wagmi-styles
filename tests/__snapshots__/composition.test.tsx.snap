// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`composition composed style 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
      },
      Object {
        "height": 100,
        "width": 200,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
    ]
  }
/>
`;

exports[`composition composed style 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
      },
      Object {
        "height": 200,
        "width": 400,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
    ]
  }
/>
`;

exports[`composition composed style with theme vars 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "blue",
      },
      Object {
        "height": 100,
        "width": 200,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
    ]
  }
/>
`;

exports[`composition composed style with theme vars 2`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
      },
      Object {
        "height": 200,
        "width": 400,
      },
      Object {
        "margin": 8,
        "padding": 16,
      },
    ]
  }
/>
`;
